src/app/service.rs

```diff
@@ -38,31 +37,33 @@ impl<T: Send + Sync> Service<http::Request<Incoming>> for AppService<T> {
     type Response = http::Response<ResponseBody>;
 
     fn call(&self, request: http::Request<Incoming>) -> Self::Future {
-        let mut params = PathParams::new(Vec::with_capacity(7));
-        let mut next = Next::new(VecDeque::with_capacity(7));
+        let mut params = PathParams::new(Vec::with_capacity(8));
+        let mut next = Next::new(VecDeque::new());
 
         let path = request.uri().path();
-        let results = match self.app.router.visit(path) {
-            Err(error) => return ServeRequest { result: Err(error) },
-            Ok(visted) => visted,
-        };
 
-        for binding in &results {
-            for kind in binding.nodes() {
-                params.extend(match kind {
-                    MatchKind::Edge(MatchCond::Partial(node)) => {
-                        next.extend(node.route().filter_map(MatchCond::as_partial).cloned());
-                        node.param(|| binding.range())
-                    }
-                    MatchKind::Edge(MatchCond::Exact(node)) => {
-                        next.extend(node.route().map(MatchCond::as_either).cloned());
-                        node.param(|| binding.range())
+        for binding in self.app.router.visit(path) {
+            for node in binding.results() {
```

Comment:
While returning references to nodes from via-router could potentially expose the entire route tree to a hacker, the references are short lived enough to not make a difference.

If anyone disagrees, please comment and provide an explanation.

As far as I can tell, as long as the references are not live between await points in the code, you would have to be the luckiest person on the planet to get a reference back to the route tree while the server is still running.